Sources/Valet/SecureEnclaveValet.swift

```diff
@@ -226,13 +254,90 @@ public final class SecureEnclaveValet: NSObject {
 
     // MARK: Internal Properties
 
+    /// A provider of an `LAContext` used when needing to manually evaluate an authentication policy
+    /// prior to retrieving a value from the keychain in order to allow the evaluation prompt to show a
+    /// custom fallback button to the user.  Should always return a new instance of `LAContext` in
+    /// practice, but specified as an internal var to allow it to be overridden in tests.
+    #if !os(watchOS)
+    @available(tvOS 10.0, *)
+    internal lazy var authenticationContextProvider: () -> LAContext = { LAContext() }
+    #endif
+
     internal let service: Service
 
     // MARK: Private Properties
 
     private let lock = NSLock()
     private let baseKeychainQuery: [String : AnyHashable]
 
+    // MARK: - Private Methods
+
+    /// Synchronously calls `LAContext.evaluatePolicy`, passing the evaluated context
+    /// via `kSecUseAuthenticationContext` when querying for the string at the given
+    /// key in the secure enclave.
+    #if !os(watchOS)
+    @available(tvOS 10.0, *)
+    private func synchronouslyEvaluatePolicy(
+        forKey key: String,
+        withPrompt userPrompt: String,
+        withFallbackTitle fallbackTitle: String
+    ) throws -> String {
+        // Use a semaphore to block the thread and perform the evaluation synchronously.
+        let semaphore = DispatchSemaphore(value: 0)
+        var result: Result<String, Error>?
+
+        let authenticationContext = authenticationContextProvider()
+        authenticationContext.localizedFallbackTitle = fallbackTitle
+        authenticationContext.evaluatePolicy(
+            accessControl.policy,
+            localizedReason: userPrompt,
+            reply: { success, error in
+                if success {
+                    do {
+                        var keychainQuery = self.baseKeychainQuery
+                        keychainQuery[kSecUseAuthenticationContext as String] = authenticationContext
+                        let string = try SecureEnclave.string(
+                            forKey: key,
+                            withPrompt: userPrompt,
+                            options: keychainQuery
+                        )
+                        result = .success(string)
+
+                    } catch {
+                        result = .failure(error)
+                    }
+
+                } else if let error = error as? LAError {
+                    let transformedError = LAErrorTransformer.transform(
+                        error: error,
+                        accessControl: self.accessControl
+                    )
+                    switch transformedError {
+                    case let .keychain(error):      result = .failure(error)
+                    case let .secureEnclave(error): result = .failure(error)
```

Comment:
I think this only works if both of the associated values were the same type

Reply:
ahh yeah only if they're both `Error`. Otherwise you'll need:
```suggestion
                    case let .keychain(error as Error),
                             let .secureEnclave(error as Error):
                        result = .failure(error)
```